hastory-cli/src/Hastory/Cli/Commands/Gather.hs

```diff
@@ -1,29 +1,58 @@
 {-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE OverloadedStrings #-}
 
 module Hastory.Cli.Commands.Gather where
 
-import Import
+import Control.Monad.Catch
+import Control.Monad.Except (runExceptT)
+import Control.Monad.Extra (whenJustM)
+import Control.Monad.IO.Unlift (MonadUnliftIO)
+import Control.Monad.Logger (MonadLogger)
+import Control.Monad.Logger.CallStack (logWarn, runStdoutLoggingT)
+import Control.Monad.Reader
+import Data.Semigroup ((<>))
+import Data.Text (Text)
+import qualified Data.Text as T
+import qualified Data.Text.IO as T
+import qualified Database.Persist.Sqlite as SQL
 
 import Data.Hastory
+import Data.Hastory.API
 import Hastory.Cli.Internal
 import Hastory.Cli.OptParse.Types
 
-import Data.Text (Text)
-
-import qualified Data.Text.IO as T
-import qualified Database.Persist.Sqlite as SQL
-
 gather :: (MonadReader Settings m, MonadThrow m, MonadUnliftIO m) => m ()
 gather = do
-    text <- liftIO T.getContents
-    gatherFrom text
+  text <- liftIO T.getContents
+  gatherFrom text
 
-gatherFrom :: (MonadReader Settings m, MonadThrow m, MonadUnliftIO m) => Text -> m ()
+gatherFrom ::
+     (MonadReader Settings m, MonadThrow m, MonadUnliftIO m) => Text -> m ()
 gatherFrom text = do
-    entry <- liftIO $ gatherEntryWith text
-    storeHistory entry
+  entry <- liftIO $ gatherEntryWith text
+  storeHistory entry
+
+sendEntryToStorageServer ::
+     (MonadIO m, MonadLogger m) => Entry -> RemoteStorageClientInfo -> m ()
+sendEntryToStorageServer entry (RemoteStorageClientInfo url token) =
+  runExceptT (mkHastoryClient url token) >>= \case
+    Left err ->
+      logWarn $
+      "Couldn't create the remote storage client: " <> T.pack (show err)
+    Right client -> do
+      resp <-
+        liftIO $ runHastoryClientM client $ \t -> appendCommand (Just t) entry
+      case resp of
+        Left err ->
+          logWarn $
+          "Saving command in remote storage server has failed: " <>
+          T.pack (show err)
+        Right _ -> pure ()
 
 storeHistory ::
-       (MonadReader Settings m, MonadThrow m, MonadUnliftIO m) => Entry -> m ()
-storeHistory = runDb . SQL.insert_
+     (MonadReader Settings m, MonadThrow m, MonadUnliftIO m) => Entry -> m ()
+storeHistory entry = do
+  _key <- runDb (SQL.insert entry)
+  whenJustM (asks remoteStorageClientInfo) $
```

Comment:
What happens if this fails because the internet is down? Do you get a weird status code in your terminal?

Reply:
ServantError would be thrown but it's catched by `sendEntryToStorageServer` and converted into a warning log. In the most recent iteration we started ignoring these warning logs, until we find a user-friendly way of directing them.